platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -688,6 +857,8 @@ namespace platoon_strategic_ihp
                 request.location = pose_to_ecef(pose_msg_);
```

Comment:
Assignment of plan_id above should use the singleton "gen" member variable I suggested above.  Look for other random generation statements elsewhere to make sure they all do it the same way.